Polyfills/Console/Source/Console.cpp

```diff
@@ -34,16 +34,82 @@ namespace
         }
     }
 
+    inline bool isNumberEnding(char c)
+    {
+        return c == 'd' || c == 'i' || c == 'f';
+    }
+
+    inline bool isStringOrObjectEnding(char c)
+    {
+        return c == 'o' || c == 'O' || c == 's';
+    }
+
     void InvokeCallback(Babylon::Polyfills::Console::CallbackT callback, const Napi::CallbackInfo& info, Babylon::Polyfills::Console::LogLevel logLevel)
     {
-        std::stringstream ss{};
-        for (size_t i = 0; i < info.Length(); i++)
+        std::ostringstream ss{};
+        if (info.Length() > 0)
         {
-            if (i > 0)
+            std::string firstArg = info[0].ToString();
+            size_t currArgIndex = 1;
+
+            std::size_t j = 0;
+            while (j < firstArg.size())
+            {
+                const char currChar = firstArg.at(j);
```

Comment:
Use `[]` here since you are already doing a bounds check. `at` will perform bounds check that is unnecessary. https://stackoverflow.com/questions/14699060/what-is-the-difference-between-stringat-and-stringoperator